### DISCUSSION TITLE
migration because of problem with Matter

### DISCUSSION THREAD

**Discussion by thebadboyltz** [technyon/nuki_hub#532] [Q&A]
Hello,
Following many problems with Matter, I finally decided to switch to your NuKiHub (thank you for your great work).

I installed the 9.01 software on an esp32 ETH01 which is connected to the LAN by cable, but I have not yet paired my NuKi because I prefer to read and understand all the many parameters there are.

However, I have several questions.
To have something as fast as my current configuration (NuKi-mqtt --> Matter(appletv)-->HA) I will go with the hybrid mode.
If I understand correctly with the hybrid mode, the basic mqtt section must remain empty since it is mqtt of the NuKi firmware that takes over?.

By the way, how does the hybrid mode work
- NUKI-mqtt–>BLE esp32–>LAN–>HA?

With hybrid mode, do we recover all entities on HA in the same way as NuKiHub-->BLE?

Thanks

**Comment by technyon** [comment]
You still have to configure the broker in Nuki Hub if thats your question.

**Comment by thebadboyltz** [comment]
ok thanks, if I understand correctly the hybrid mode will still use thread and matter but can also work with bluetooth if the matter network is down?

My settings are good like this, sorry to ask stupid questions but it seems complicated for my level even if the explanations are excellent

I have trouble understanding this explanation in hybrid mode


Optionally enable Allow locking. Note that if you enable this setting it is preferred to set ACL on your MQTT broker to only allow the Nuki lock and Nuki Hub MQTT user access to the topic nuki/NUKI-ID/lockAction to make sure that only Nuki Hub can execute commands on the lock (otherwise ACL settings through Nuki Hub can not be 100% enforced)


![unnamed](https://github.com/user-attachments/assets/899b845d-4d99-46d3-87a0-c6f35a2f35f3)

**Comment by technyon** [comment]
The hybrid mode has nothing to do with thread or matter, that works seperately from Nuki Hub. The hybrid mode combines the MQTT API of a Pro lock with the MQTT API of Nuki Hub, so you kind of get the best of both worlds. You have to enable the MQTT API of your lock and configure MQTT on Nuk Hub.

**Comment by thebadboyltz** [comment]
ok, but to activate mqtt on the API NUKI must be connected to the network, so I still have to activate Matter or wifi?

Sorry if for you all seems logical to me everything is not yet clear.

For the moment here are the steps I did.

- pair NUKI with the API (by pressing 10 seconds on NUKI)
-pair NUKI with NuKiHub
-activate Matter to have access to the MQTT configuration of the API, the API is well connected (green flesh) and I have deactivated auto-discovery)

then I do not see that the hybrid mode is activated and the NUKI goes up 2x in HA


![png-clipart-home-cinema-electronics-hard-copy-icon-cartoon-sound-and-tv-cartoon-character-television-thumbnail](https://github.com/user-attachments/assets/be0336b7-0f39-44d9-87b1-1df01d1a0f34)
![60394fa9b6264f0004079c1a](https://github.com/user-attachments/assets/9c0ebf52-045b-4745-b8ca-62b6b4b47a06)
![Capture](https://github.com/user-attachments/assets/1bd72aa5-754d-42e3-8107-f751ae554e73)
![apple_tv_logo](https://github.com/user-attachments/assets/2d157fa6-e03a-4a8a-9382-7763623e6d26)
![Captur](https://github.com/user-attachments/assets/01c424a3-8af6-4b5b-ae1c-2b78aae94695)
![unnamed](https://github.com/user-attachments/assets/3affe562-dc3b-489a-bcfe-ba3a1563793e)

**Reply by iranl** [reply]
> ok, but to activate mqtt on the API NUKI must be connected to the network, so I still have to activate Matter or wifi?

Yes, your Nuki device must be connected to the network over Matter-over-Thread or WiFi (Matter-over-Thread is prefered) and MQTT on the Nuki device must be setup correctly.

 
> then I do not see that the hybrid mode is activated and the NUKI goes up 2x in HA

I'm not sure if Nuki removes their auto discovery MQTT topics when you disable auto-discovery after having it enabled previously.
You can safely manually remove the Nuki device from Home Assistant.

I do highly recommend updating to 9.02 beta as soon as possible, as release 9.01 was not a very stable release. You can update using the built in firmware updater.

If hybrid mode is enabled you should see it enabled on the homepage of NukiHub and on the info page (click the version number behind "Firmware" on the Nuki hub homepage)

To answer some of your previous questions about Hybrid mode:

-Both NukiHub and the Nuki device need to be connected to your network and be configured to use the same MQTT broker.
-NukiHub will subscribe to the MQTT topics that the Nuki device publishes. When the Nuki device publishes information (such as a changed lockstate) to its MQTT topics NukiHub will detect this and proxy this information to it's own MQTT topics. This information will than be picked up by the HomeAssistant device that is created by NukiHub. This is the main advantage of hybrid mode as state changes will be picked up much faster than when using bluetooth.
-If NukiHub wants to perform an action on the Nuki device (lock/unlock etc. through HomeAssistant) it will not use bluetooth but will also proxy the request by publishing the requested action to the Nuki device MQTT action topic. This again is faster than bluetooth.
-Any functionality that is not available in the MQTT API but is available in NukiHub (almost anything you can do in the Nuki app) is done over bluetooth. This includes requesting and changing/addingremoving logs, keypad codes, timecontrol, all Nuki device settings, rebooting the Nuki device etc. NukiHub in Hybrid mode works almost the same as in regular mode, but lock state changes and lock actions through NukiHub are a lot faster.

All of the above should give you the best of all worlds in regards to speed, battery, reliability and functionality

If supported by the Nuki MQTT API (lock state/actions): Nuki device <-> WiFi/Thread <-> MQTT <-> NukiHub <->WiFi/LAN <->MQTT <–> HA
If not supported by the Nuki MQTT API (all other NukiHub functionality): Nuki device <-> BLE <-> NukiHub <->WiFi/LAN <->MQTT <–> HA

**Comment by thebadboyltz** [comment]
thank you, your explanations are excellent, I finally understood.

I upgraded to firmware beta 9.02, if my Matter network is down, how do I do an HA automation to restart the lock?

last question, I put 3 seconds in the query interval, does this mean that NuKiHub will connect every 3 seconds in BLE to retrieve the status even if the Matter connection is good? or if the Matter connection is down at this time it launches queries every 3 seconds.

thank you for your support

![Capture](https://github.com/user-attachments/assets/8a21b01a-e7fe-448b-8a4a-249bb7cd1dd0)

**Reply by iranl** [reply]
> I upgraded to firmware beta 9.02, if my Matter network is down, how do I do an HA automation to restart the lock?

Enable the Reboot Nuki setting on the bottom of the NukiHub Access Level Configuration (only available starting in 9.02, disabled by default). 

<img width="572" alt="image" src="https://github.com/user-attachments/assets/a2ae58f2-e8dc-4d06-acfc-a0b9e73e5107">

After saving reboot NukiHub to create a button in Home Assistant to reboot the Nuki device.

<img width="239" alt="image" src="https://github.com/user-attachments/assets/465fa7dd-2641-40e1-8f11-13ec49042f4b">

Example automation to reboot the Nuki device if MQTT on the Nuki device fails for >3 minutes

```
description: ""
mode: single
triggers:
  - trigger: state
    entity_id:
      - binary_sensor.nukilock_hybrid_connected
    to: "off"
    for:
      hours: 0
      minutes: 3
      seconds: 0
    from: "on"
conditions:
  - condition: and
    conditions:
      - condition: state
        entity_id: binary_sensor.nukilock_hybrid_connected
        state: "on"
      - condition: state
        entity_id: binary_sensor.nukihub_mqtt_connected
        state: "on"
actions:
  - action: button.press
    target:
      entity_id: button.nukilock_reboot_nuki
    data: {}
  - delay:
      hours: 0
      minutes: 10
      seconds: 0
      milliseconds: 0
```
 
> last question, I put 3 seconds in the query interval, does this mean that NuKiHub will connect every 3 seconds in BLE to retrieve the status even if the Matter connection is good? or if the Matter connection is down at this time it launches queries every 3 seconds.

Yes, don't do this, it completely defeats the purpose of hybrid mode. Put the top 3 intervals at 86400 as suggested in the [Hybrid mode README](https://github.com/technyon/nuki_hub/blob/master/HYBRID.md). You can set the query interval when hybrid mode is down on the "Network Configuration" page of NukiHub

**Comment by thebadboyltz** [comment]
I am ashamed to ask you stupid questions when the answers are simple.
Here I have the restart NUKI button but it has no effect on my NUKI beta version 4.5, however with version 9.0.2 when I want to go to the network settings, it does not open an interface but asks me to save a file named "ntwconfig".
If I want to modify the query interval, I have to open the file with notepad, then send it with import-export configuration?

I also saw that it was possible to set a ringtone at the opening or closing, where do I have to go to activate these settings?

When NuKi is charging, we do not know how to recover the charging state?

thank you

**Reply by iranl** [reply]
> Here I have the restart NUKI button but it has no effect on my NUKI beta version 4.5, however with version 9.0.2 when I want to go to the network settings, it does not open an interface but asks me to save a file named "ntwconfig".
> If I want to modify the query interval, I have to open the file with notepad, then send it with import-export configuration?

Reboot Nuki only works if you configure your Nuki PIN in NukiHub (Credentials page, with the big orange "Please configure PIN" on the home page)

In regards to not opening the WebInterface this is probably some caching issue in your browser. Try incognito mode or another browser

> I also saw that it was possible to set a ringtone at the opening or closing, where do I have to go to activate these settings?

There is no such feature for the Lock. You are probably looking at a feature for the Nuki Opener

> When NuKi is charging, we do not know how to recover the charging state?

I don't think I understand the question.

**Comment by thebadboyltz** [comment]
the pin code is set correctly but the reboot does not work (My NUKI is in beta 4.5) it may be a problem.

For the opening of the network configuration page, I tested from 3 browsers and each time it wants to download a file instead of opening the web page.

I reinstalled in the esp32, then I put back the version 9.02, at this time the network configuration page opens normally. I uploaded my backup parameters (database + coupling) and after the configuration page no longer opens and offers me to download the file

I upgraded to version 9.0.2 and it does not change anything.

My question for the battery, Before migrating to NuKiHub, I could see the battery status in HA, either charging or not charging.

With NuKiHub, when the battery is charging I do not see it in the HA entities.

[
![Capture](https://github.com/user-attachments/assets/db83238f-49ef-4c31-9d74-491bac864522)
](url)



I don't remember having activated register as an APP as recommended to save battery, but I see that it is still activated

![Capture](https://github.com/user-attachments/assets/8343f3f3-df12-410e-ac31-f2fd9df2c7a5)

**Reply by iranl** [reply]
> the pin code is set correctly but the reboot does not work (My NUKI is in beta 4.5) it may be a problem.

Should be fixed in 9.03

> My question for the battery, Before migrating to NuKiHub, I could see the battery status in HA, either charging or not charging.

We do not have a Home Assistant auto discovery entity for charging state at the moment.
I've added it in #546. This will be in 9.03-master4

> I don't remember having activated register as an APP as recommended to save battery, but I see that it is still activated

In Hybrid mode NukiHub has to be registered as app as using a bridge device with a Nuki Lock disables MQTT/Network connectivity in the Nuki device (unnecessary design choice by Nuki). Register as app is automatically enabled when enabling hybrid mode.

NukiHub works perfectly fine in app mode in hybrid mode and the battery argument for using bridge mode does not apply for hybrid mode.

**Comment by thebadboyltz** [comment]
The status of the battery to know if it is charging or not, is not visible in the HA entities, but the information goes back to mqtt

![png-clipart-home-cinema-electronics-hard-copy-icon-cartoon-sound-and-tv-cartoon-character-television-thumbnail](https://github.com/user-attachments/assets/c7b86081-b112-4d5d-8b59-e61129c2c47a)
![Capture](https://github.com/user-attachments/assets/a82e74af-8b7e-4f06-a9b6-a9f991e4988d)
